lib/src/native/realm_core.dart

```diff
@@ -116,26 +150,146 @@ class _RealmCore {
+  }
+
+  void closeRealm(Realm realm) {
+    _realmLib.invokeGetBool(() => _realmLib.realm_close(realm.handle._pointer), "Realm close failed");
+  }
+
+  void beginWrite(Realm realm) {
+    _realmLib.invokeGetBool(() => _realmLib.realm_begin_write(realm.handle._pointer), "Could not begin write");
+  }
+
+  void commitWrite(Realm realm) {
+    _realmLib.invokeGetBool(() => _realmLib.realm_commit(realm.handle._pointer), "Could commit write");
+  }
+
+  bool getIsWritable(Realm realm) {
+    return _realmLib.realm_is_writable(realm.handle._pointer);
+  }
+
+  void rollbackWrite(Realm realm) {
+    _realmLib.invokeGetBool(() => _realmLib.realm_rollback(realm.handle._pointer), "Could rollback write");
+  }
+
+  int getClassKey(Realm realm, String className) {
+    return using((Arena arena) {
+      Pointer<Uint8> found = arena<Uint8>();
+      Pointer<realm_class_info_t> classInfo = arena<realm_class_info_t>();
+      _realmLib.invokeGetBool(() => _realmLib.realm_find_class(realm.handle._pointer, className.toUtf8Ptr(arena), found, classInfo),
+          "Error getting class $className from realm at ${realm.config.path}");
+
+      if (found.value == 0) {
+        final error = getLastError();
+        throw RealmException("Class $className not found in ${realm.config.path}. Error: $error");
+      }
+
+      return classInfo.ref.key;
+    });
+  }
+
+  Map<String, int> getPropertyKeys(Realm realm, int classKey) {
+    return using((Arena arena) {
+      Pointer<IntPtr> propertyCountPtr = arena<IntPtr>();
+      _realmLib.invokeGetBool(
+          () => _realmLib.realm_get_property_keys(realm.handle._pointer, classKey, nullptr, 0, propertyCountPtr), "Error getting property count");
+
+      var propertyCount = propertyCountPtr.value;
+      final propertiesPtr = arena<realm_property_info_t>(propertyCount);
+      _realmLib.invokeGetBool(() => _realmLib.realm_get_class_properties(realm.handle._pointer, classKey, propertiesPtr, propertyCount, propertyCountPtr),
+          "Error getting class properties.");
+
+      propertyCount = propertyCountPtr.value;
+      Map<String, int> result = <String, int>{};
+      for (var i = 0; i < propertyCount; i++) {
+        final property = propertiesPtr.elementAt(i);
+        result[property.ref.name.cast<Utf8>().toDartString()] = property.ref.key;
+      }
+      return result;
+    });
+  }
+
+  RealmObjectHandle createRealmObject(Realm realm, int classKey) {
+    final realmPtr = _realmLib.invokeGetPointer(() => _realmLib.realm_object_create(realm.handle._pointer, classKey));
+    return RealmObjectHandle._(realmPtr);
+  }
+
+  Object readProperty(RealmObject object, int propertyKey, RealmPropertyType propertyType) {
+    return using((Arena arena) {
+      Pointer<realm_value_t> value = arena<realm_value_t>();
+      _realmLib.invokeGetBool(() => _realmLib.realm_get_value(object.handle._pointer, propertyKey, value));
+
+      switch (propertyType) {
+        case RealmPropertyType.int:
```

Comment:
Reminds of a thing I intended to discuss.. Should we just call it `RealmType` instead of `RealmPropertyType`?

Reply:
yes we could do that, I just thought at the time that this is more descriptive

Reply:
also this is how core c-api names these values. I suggest we leave it as RealmPropertyType. And then there will be RealmCollectionType